dbt/loader.py

```diff
@@ -217,6 +219,17 @@ def load_project(cls, root_project, all_projects, project, project_name,
             macros=macros)
 
 
+class DocumentationLoader(ResourceLoader):
+    @classmethod
+    def load_project(cls, root_project, all_projects, project, project_name,
+                     macros):
+        return dbt.parser.DocumentationParser.load_and_parse(
+            package_name=project_name,
+            root_project=root_project,
+            all_projects=all_projects,
+            root_dir=project.get('project-root'),
+            relative_dirs=project.get('source-paths', []))
```

Comment:
Hmm, I didn't even consider that. I just immediately put it next to the schema.yml. But it does make some sense.

If we did so maybe we should consider something like: `relative_dirs=project.get('source-paths', []) + project.get('docs-paths', [])`

I actually also wouldn't be against also searching from the root directory, if that's feasible with dbt (`maybe relative_dirs + ['.']` would work?).

Reply:
@beckjake I was thinking the same thing around searching the root directory! I'm a little concerned thought that a random .md file could contain code that would break dbt.

Just as an example, a `styleguide.md` file (like [this](https://github.com/fishtown-analytics/corp/blob/master/dbt_coding_conventions.md) might contain `{{ ref(...) }}` in example sql, and I think the docs parser would probably choke on that, right?

After some testing, we do a recursive search through the dirs, so including `.` in the list of paths will also snag the `target/` directory, which is probably not what we want here.

I think the right way to handle this is to just use `docs-paths`, where the default value is:
```
"docs-paths": ["models"]
```

This will let users override the dirs in their `dbt_project.yml` file, but will still be sort of a sane default value. You buy that?

Reply:
I mostly agree, but I think we should default `docs-paths` to the contents of `source-paths`, rather than `["models"]`, if we can. It's a subtle difference, but the idea of "by default we search your source paths" seems really compelling to me.

Reply:
cool, I buy that. Let's make sure to document that quirk in the docs for the docs :)